### ISSUE TITLE
Terminating Pending workflows waiting on a mutex/semaphore are not immediate

### ISSUE THREAD

**Issue by jessesuen** [argoproj/argo-workflows#10733]
### Pre-requisites

- [X] I have double-checked my configuration
- [X] I can confirm the issues exists when I tested with `:latest`
- [X] I'd like to contribute the fix myself (see [contributing guide](https://github.com/argoproj/argo-workflows/blob/master/docs/CONTRIBUTING.md))

### What happened/what you expected to happen?

After submitting a workflow that becomes Pending because it is waiting on a mutex or sempahore, when that workflow is terminated, it will not be terminated immediately, even though it has yet to begun. It will only be terminated once that workflow can acquire the semaphore/mutex, which may be a long time because it needs to wait for it to be released by another workflow.

We should allow Pending workflows to terminate immediately if they never had aquired the semaphore, because there is no consequence.

To reproduce:
1. Submit two workflows in example below
2. Run `argo terminate` on second workflow that is in Pending state:
3. Notice that the Pending workflow is still waiting on lock:

```
$ argo terminate synchronization-wf-level-cl7rp
workflow synchronization-wf-level-cl7rp terminated

$ argo list
NAME                             STATUS    AGE   DURATION   PRIORITY   MESSAGE
synchronization-wf-level-cl7rp   Pending   11s   0s         0          Waiting for argo/Mutex/test lock. Lock status: 0/1
synchronization-wf-level-rcnqz   Running   12s   12s        0
```

To fix this, I think the controller need to consider examining the following when reconciling Pending workflows waiting on mutexes/semaphores:

```yaml
spec:
  shutdown: Terminate
```

And proceed with the termination even if the workflow is not at the top of the priority queue. 

### Version

v3.4.5

### Paste a small workflow that reproduces the issue. We must be able to run the workflow; don't enter a workflows that uses private images.

```YAML
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: synchronization-wf-level-
spec:
  entrypoint: whalesay
  synchronization:
    mutex:
      name:  test
  templates:
    - name: whalesay
      container:
        image: docker/whalesay:latest
        command: [sh, -c]
        args: ["sleep 99999"]
```

### Logs from the workflow controller

```text
time="2023-03-22T17:22:19.611Z" level=info msg="Processing workflow" namespace=argo workflow=synchronization-wf-level-cl7rp
time="2023-03-22T17:22:19.620Z" level=warning msg="Workflow processing has been postponed due to concurrency limit" namespace=argo workflow=synchronization-wf-level-cl7rp
time="2023-03-22T17:22:19.620Z" level=info msg="Updated phase  -> Pending" namespace=argo workflow=synchronization-wf-level-cl7rp
time="2023-03-22T17:22:19.620Z" level=info msg="Updated message  -> Waiting for argo/Mutex/test lock. Lock status: 0/1" namespace=argo workflow=synchronization-wf-level-cl7rp
time="2023-03-22T17:22:19.626Z" level=info msg="Workflow update successful" namespace=argo phase=Pending resourceVersion=366960480 workflow=synchronization-wf-level-cl7rp
time="2023-03-22T17:22:38.098Z" level=info msg="Processing workflow" namespace=argo workflow=synchronization-wf-level-cl7rp
time="2023-03-22T17:22:38.098Z" level=warning msg="Workflow processing has been postponed due to concurrency limit" namespace=argo workflow=synchronization-wf-level-cl7rp
```

### Logs from in your workflow's wait container

```text
N/A
```

**Comment by jessesuen** [comment]
Note that **_deleting_** Pending workflows are not affected and happen immediately. Only **_terminating_** pending workflows.
